### PR TITLE
Manage notification contacts

### PR DESCRIPTION
**Pull request by saliceti**
# What

We want to be able to create and manage notification contacts and set pingdom notification alerts via [terraform](https://github.com/russellcardullo/terraform-provider-pingdom).

Note this only works for multi-user accounts.

# How to review

You can play with this example code for example:

```go
	client := pingdom.NewMultiUserClient("username","password", "app key", "account email")
	url, _ := url.Parse("https://api.pingdom.com")
	client.BaseURL = url

	// List contacts
	contacts, err := client.Contacts.List()
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(contacts)

	// Create contact
	newContact := pingdom.Contact{
		Name:               "Test " + strconv.Itoa(rand.Int()),
		Email:              "test@site.com",
		Defaultsmsprovider: "nexmo",
	}

	contactResponse, err := client.Contacts.Create(&newContact)
	if err != nil {
		fmt.Println(err)
	}

	contactID := contactResponse.ID

	contacts, err = client.Contacts.List()
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(contacts)

	// Create check with contact ID
	newCheck := pingdom.HttpCheck{
		Name:                     "HTTP check " + strconv.Itoa(rand.Int()),
		Hostname:                 "example.com",
		Resolution:               1,
		ContactIds:               []int{contactID},
		SendNotificationWhenDown: 2,
                UseLegacyNotifications: true,
                SendToEmail: true,
	}
	checkResponse, err := client.Checks.Create(&newCheck)
	if err != nil {
		fmt.Println(err)
	}

	check, err := client.Checks.Read(checkResponse.ID)
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(check)

	_, err = client.Checks.Delete(checkResponse.ID)
	if err != nil {
		fmt.Println(err)
	}
	// Update contact
	updateContact := pingdom.Contact{
		Name:               "Test " + strconv.Itoa(rand.Int()),
		Email:              "test@site.com",
		Defaultsmsprovider: "nexmo",
	}
	_, err = client.Contacts.Update(contactID, &updateContact)
	if err != nil {
		fmt.Println(err)
	}

	contacts, err = client.Contacts.List()
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(contacts)

	// Delete contact
	_, err = client.Contacts.Delete(contactResponse.ID)
	if err != nil {
		fmt.Println(err)
	}

	contacts, err = client.Contacts.List()
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(contacts)
```

### CODE REVIEW

File: pingdom/contact_types.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	Name               string `json:"name"`
+	Email              string `json:"email,omitempty"`
+	Cellphone          string `json:"cellphone,omitempty"`
+	Countryiso         string `json:"countryiso,omitempty"`
+	CountryCode        string `json:"countrycode,omitempty"`
+	Defaultsmsprovider string `json:"defaultsmsprovider,omitempty"`
+	Directtwitter      bool   `json:"directtwitter,omitempty"`
+	Twitteruser        string `json:"twitteruser,omitempty"`
+	Iphonetokens       string `json:"iphonetokens,omitempty"`
+	Androidtokens      string `json:"androidtokens,omitempty"`
```
Comment (russellcardullo) [inline pingdom/contact_types.go]: Looks great!  Would you mind updating some of the capitalization for a few of these fields to improve readability as well as for consistency with how we've capitalized other fields? I was thinking the following ones:
```
CountryISO
DefaultSMSProvider
DirectTwitter
TwitterUser
IphoneTokens
AndroidTokens
```

File: pingdom/api_responses.go [hunk 2]
```diff
@@ -46,6 +46,21 @@ type CheckResponseType struct {
...
+	ID                 int    `json:"id"`
+	Name               string `json:"name"`
+	Email              string `json:"email,omitempty"`
+	Cellphone          string `json:"cellphone,omitempty"`
+	Countryiso         string `json:"countryiso,omitempty"`
+	Defaultsmsprovider string `json:"defaultsmsprovider,omitempty"`
+	Directtwitter      bool   `json:"directtwitter,omitempty"`
+	Twitteruser        string `json:"twitteruser,omitempty"`
+	Iphonetokens       string `json:"iphonetokens,omitempty"`
+	Androidtokens      string `json:"androidtokens,omitempty"`
```
Comment (russellcardullo) [inline pingdom/api_responses.go]: Would you mind capitalizing these as well to be consistent with capitalization in types?

**Comment by russellcardullo** [comment]
Thanks for the well documented PR!  I tested it out and everything works for me.  The only thing I'd suggest is we fix the capitalization on some of the field names to be consistent as per the inline comments I made.

I noticed that you said `this only works for multi-user accounts`.  I tested it with a single user account and it worked for me in that context.  Let me know if there's anything that might not work for a single user account.

**Comment by saliceti** [comment]
This is the issue:

```bash
$ curl -H "App-Key: XXXXXX" -X POST -d "name=toto" https://XXXXXX:XXXXXX@api.pingdom.com/api/2.0/contacts
{"error":{"statuscode":410,"statusdesc":"Gone","errormessage":"Method is deprecated, see notification_contacts"}}
```

```bash
$ curl -H "App-Key: XXXXXX" -X POST -d "name=toto" https://XXXXXX:XXXXXX@api.pingdom.com/api/2.0/notification_contacts
{"error":{"statuscode":410,"statusdesc":"Gone","errormessage":"Method is deprecated"}}
```

The problem is actually not related to multi-user or not. Something has changed in the API for **new accounts**. I have contacted pingdom support and this is what they've said:

>What we think causes this is likely due to some small differences in the alerting types. Newer accounts have a new set of alerts that work in a different way. 
>This of course needs to be fixed and we'll need to update the docs. As soon as it's fixed I'll let you know!

I will remove the warning from the PR. You can then decide to merge it or wait for the issue to be resolved by pingdom for new accounts. I would let you know when they come back to me.

**Comment by russellcardullo** [comment]
Thanks for the note about the issue with new accounts.  Let's merge this now and if we hear back from Pingdom we can decide how to address that issue.

**Pull request merged**